lib/fluent/plugin/out_s3.rb

```diff
@@ -76,19 +78,15 @@ def start
       if @aws_key_id && @aws_sec_key
         options[:access_key_id] = @aws_key_id
         options[:secret_access_key] = @aws_sec_key
-      elsif ENV.key? "AWS_ACCESS_KEY_ID"
-        options[:credential_provider] = AWS::Core::CredentialProviders::ENVProvider.new('AWS')
-      else
-        options[:credential_provider] = AWS::Core::CredentialProviders::EC2Provider.new({:retries => @aws_iam_retries})
```

Comment:
I believe the original intent here was to have configurable retries for retrieving credentials, because the ec2 metadata service sometimes does not respond reliably. Although `:credential_provider` is no longer available in sdk v2, users can still construct `:credentials` (such as `Aws::InstanceProfileCredentials`) directly with `:retries`. See http://docs.aws.amazon.com/sdkforruby/api/Aws/InstanceProfileCredentials.html

We'll either have to implement something similar to the default provider chain with configurable retries, or drop support for `@aws_iam_retries` from this plugin's options.